tools/clang/lib/SPIRV/LowerTypeVisitor.cpp

```diff
@@ -547,8 +547,10 @@ const SpirvType *LowerTypeVisitor::lowerResourceType(QualType type,
 
     // Calculate memory alignment for the resource.
     uint32_t size = 0, stride = 0;
+    QualType sArray = astContext.getConstantArrayType(
+        s, llvm::APInt(32, 1), clang::ArrayType::Normal, 0);
     std::tie(std::ignore, size) =
-        alignmentCalc.getAlignmentAndSize(s, rule, isRowMajor, &stride);
+        alignmentCalc.getAlignmentAndSize(sArray, rule, isRowMajor, &stride);
```

Comment:
The `stride` of `sArray` which is an array stride is being used as matrix stride on line 563.

Reply:
Notice that in line 557 below:
```cpp
const auto *raType = spvContext.getRuntimeArrayType(structType, size);
```
the second argument that `getRuntimeArrayType` takes is the `ArrayStride` for the runtime array. I think it should now be updated to take `stride` instead of `size` ?

Reply:
The root cause of this issue is that there is no representation for `RuntimeArray` as a QualType.

I think one solution could be to call `getAlignmentAndSize` twice.
1) call `getAlignmentAndSize` with `sArray`. --> always use the `stride` result of this call as the ArrayStride of the runtime array.
2) **if `s` is a matrix**, call `getAlignmentAndSize` with `s`. --> use the `stride` result of this call as MatrixStride on line 563.

Reply:
Yeah, I am on the exactly same page. I updated it as you mentioned and checked it works.